model/map/src/main/java/org/keycloak/models/map/events/MapAdminEventQuery.java

```diff
@@ -90,7 +90,7 @@ public AdminEventQuery operation(OperationType... operations) {
 
     @Override
     public AdminEventQuery resourceType(ResourceType... resourceTypes) {
-        mcb = mcb.compare(SearchableFields.RESOURCE_TYPE, EQ, Arrays.stream(resourceTypes));
+        mcb = mcb.compare(SearchableFields.RESOURCE_TYPE, IN, Arrays.stream(resourceTypes).map(Enum::toString));
```

Comment:
I believe we shouldn't transform `ResouceType` to `String` at this level, the searchable field states that it expects `ResourceType.class` as a filed type. I'd suggest rather change `MapFiledsPredicates` class as mentioned in https://github.com/keycloak/keycloak/pull/12299#discussion_r888707679 but fist we should decide how the search should behave for ResourceType.CUSTOM (https://github.com/keycloak/keycloak/pull/12299#discussion_r889222201)

Reply:
Agreed, will revert this change.